### PR TITLE
System: initial deployment of new navigator class

### PR DESCRIPTION
**Pull request by rossdotparker**
**Description**
Begins the process of rolling our @SKuipers lovely new navigator class, allowing us to remove hard coded divs with id="linkTop"

**Motivation and Context**
❤️ refactoring

**How Has This Been Tested?**
Locally

### CODE REVIEW

File: modules/Behaviour/behaviour_manage_addMulti.php [hunk 1]
```diff
@@ -48,18 +54,20 @@
...
 
     $policyLink = $settingGateway->getSettingByScope('Behaviour', 'policyLink');
-    $policyLink = 'google.com';
     if (!empty($policyLink)) {
         $form->addHeaderAction('viewPolicy', __('View Behaviour Policy'))
             ->setExternalURL($policyLink);
     }
-    if (!empty($_GET['gibbonPersonID']) || !empty($_GET['gibbonFormGroupID']) || !empty($_GET['gibbonYearGroupID']) || !empty($_GET['type'])) {
-        $form->addHeaderAction('back', (empty($policyLink) ? '' : '| ') . __('Back to Search Results'))
+    if (!empty($gibbonPersonID) or !empty($gibbonFormGroupID) or !empty($gibbonYearGroupID) or !empty($type)) {
```
Comment (SKuipers) [inline modules/Behaviour/behaviour_manage_addMulti.php:61]: Your `or` refactorings are going in the wrong direction 😆

File: modules/Library/library_manage_catalog_duplicate.php [hunk 2]
```diff
@@ -57,16 +57,19 @@
             if ($step != 1 and $step != 2) {
                 $step = 1;
             }
+            
+            $urlParamKeys = array('name' => '', 'gibbonLibraryTypeID' => '', 'gibbonSpaceID' => '', 'status' => '', 'gibbonPersonIDOwnership' => '', 'typeSpecificFields' => '');
+
+            $urlParams = array_intersect_key($_GET, $urlParamKeys);
```
Comment (SKuipers) [inline modules/Library/library_manage_catalog_duplicate.php:63]: Creative 🤔

File: modules/Reports/templates_manage_add.php [hunk 3]
```diff
@@ -37,13 +37,7 @@
...
     $page->return->setEditLink($editLink);
 
-    if ($search != '') {
-        echo "<div class='linkTop'>";
-        echo "<a href='".$gibbon->session->get('absoluteURL')."/index.php?q=/modules/Reports/templates_manage.php&search=$search'>".__('Back to Search Results').'</a>';
-        echo '</div>';
-    }
-
-    $form = Form::create('templatesManage', $gibbon->session->get('absoluteURL').'/modules/Reports/templates_manage_addProcess.php?search='.$search);
+    $form = Form::create('templatesManage', $gibbon->session->get('absoluteURL').'/modules/Reports/templates_manage_addProcess.php');
```
Comment (SKuipers) [inline modules/Reports/templates_manage_add.php:40]: Well caught 😄

**Review by SKuipers — APPROVED** [review]
A sizable chunk of refactoring, nicely done! 😄 I've made a few minor tweaks, mostly to simplify some areas that only had one variable. Looks like a fair number of edge cases, not as straight forward as we anticipated, well done catching all these. Going to merge and test 👍

**Comment by rossdotparker** [comment]
Thanks for making these final checks and tweaks, and merging : )
